cli/cmd/vm_get_ssh_endpoint.go

```diff
@@ -0,0 +1,54 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+func (r *runners) InitVMSSHEndpoint(parent *cobra.Command) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "ssh-endpoint VM_ID",
+		Short: "Get the SSH endpoint of a VM",
+		Long: `Get the SSH endpoint and port of a VM.
+
+The output will be in the format: hostname:port`,
+		Example: `# Get SSH endpoint for a specific VM by ID
+replicated vm ssh-endpoint <id>`,
+		RunE:              r.getVMSSHEndpoint,
+		Args:              cobra.ExactArgs(1),
+		ValidArgsFunction: r.completeVMIDs,
+	}
+	parent.AddCommand(cmd)
+
+	return cmd
+}
+
+func (r *runners) getVMSSHEndpoint(cmd *cobra.Command, args []string) error {
+	vmID := args[0]
+
+	vm, err := r.kotsAPI.GetVM(vmID)
+	if err != nil {
+		return errors.Wrap(err, "get vm")
+	}
+
+	if vm.DirectSSHEndpoint == "" || vm.DirectSSHPort == 0 {
+		return errors.Errorf("VM %s does not have SSH endpoint configured", vm.ID)
+	}
+
+	// Get GitHub username from API
+	githubUsername, err := r.kotsAPI.GetGitHubUsername()
+	if err != nil {
+		return errors.Wrap(err, "get github username")
+	}
+
+	// Format the SSH endpoint with username if available
+	if githubUsername != "" {
+		fmt.Printf("ssh://%s@%s:%d\n", githubUsername, vm.DirectSSHEndpoint, vm.DirectSSHPort)
+	} else {
+		fmt.Printf("ssh://%s:%d\n", vm.DirectSSHEndpoint, vm.DirectSSHPort)
```

Comment:
We should just return error here, if there's no github user affiliated with the account the user isn't going to be able to get into the box anyways because there's no key affiliated with them at that point.

We should error and point the user to updating their username to serve as a key on the Replicated portal.